### ISSUE TITLE
artemis-odb-gwt-integration-test does not compile on JDK9+

### ISSUE THREAD

**Issue by DaanVanYperen** [junkdog/artemis-odb#534]
Compiling the GWT tests on JDK10 (target 1.8) fails.

Seems related to:
https://groups.google.com/forum/#!msg/google-web-toolkit-contributors/7rDWQVWc9fU/JpW3AzX7AgAJ

Might be fixed in gwt 2.8.2, but we depend on LibGDX as well which is still on 2.8.0

Specifically:
```[INFO]    [ERROR] Hint: Check that your module inherits 'com.google.gwt.core.Core' either directly or indirectly (most often by inheriting module 'com.google.gwt.user.User')```

Full log:
```java
[INFO] java.lang.ClassNotFoundException: sun.misc.GC
[INFO] 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
[INFO] 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
[INFO] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
[INFO] 	at java.base/java.lang.Class.forName0(Native Method)
[INFO] 	at java.base/java.lang.Class.forName(Class.java:291)
[INFO] 	at com.google.gwt.dev.shell.jetty.JettyLauncher.jreLeakPrevention(JettyLauncher.java:898)
[INFO] 	at com.google.gwt.dev.shell.jetty.JettyLauncher.start(JettyLauncher.java:721)
[INFO] 	at com.google.gwt.dev.DevMode.doStartUpServer(DevMode.java:634)
[INFO] 	at com.google.gwt.dev.DevModeBase.startUp(DevModeBase.java:923)
[INFO] 	at com.google.gwt.junit.JUnitShell.getUnitTestShell(JUnitShell.java:696)
[INFO] 	at com.google.gwt.junit.JUnitShell.runTest(JUnitShell.java:670)
[INFO] 	at com.google.gwt.junit.client.GWTTestCase.runTest(GWTTestCase.java:421)
[INFO] 	at junit.framework.TestCase.runBare(TestCase.java:127)
[INFO] 	at junit.framework.TestResult$1.protect(TestResult.java:106)
[INFO] 	at junit.framework.TestResult.runProtected(TestResult.java:124)
[INFO] 	at junit.framework.TestResult.run(TestResult.java:109)
[INFO] 	at junit.framework.TestCase.run(TestCase.java:118)
[INFO] 	at com.google.gwt.junit.client.GWTTestCase.run(GWTTestCase.java:247)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at org.codehaus.mojo.gwt.test.MavenTestRunner.doRun(MavenTestRunner.java:105)
[INFO] 	at junit.textui.TestRunner.start(TestRunner.java:172)
[INFO] 	at org.codehaus.mojo.gwt.test.MavenTestRunner.main(MavenTestRunner.java:63)
[INFO]    Logging initialized @613ms
[INFO]    jetty-9.2.14.v20151106
[INFO]    [WARN] ServletContainerInitializers: detected. Class hierarchy: empty
[INFO]    Started c.g.g.j.@56c698e3{/,file:/C:/github/odb/artemis-odb/artemis-backend-gwt/artemis-gwt-test/target/www-test/,AVAILABLE}{C:\github\odb\artemis-odb\artemis-backend-gwt\artemis-gwt-test\target\www-test}
[INFO]    Started ServerConnector@675d8c96{HTTP/1.1}{0.0.0.0:61039}
[INFO]    Started @839ms
[INFO] Started web server on port 61039
[INFO] Linking modules
[INFO] Loading module 'com.ArtemisTest.JUnit'
[INFO] Public resources found in...
[INFO] Translatable source found in...
[INFO] Compiling module com.ArtemisTest.JUnit
[INFO]    Found 0 cached/archived units.  Used 0 / 0 units from cache.
[INFO]    Compiling...
[INFO]       Compilation completed in 0,00 seconds
[INFO]    PersistentUnitCache cleanup requested
[INFO]    Added 0 units to PersistentUnitCache since last cleanup
[INFO]    Skipped PersistentUnitCache because no units were added
[INFO]    Removing invalidated units
[INFO]    Tracing compile failure path for type 'java.lang.Object'
[INFO]    [ERROR] Hint: Check that your module inherits 'com.google.gwt.core.Core' either directly or indirectly (most often by inheriting module 'com.google.gwt.user.User')
[INFO] Tests run: 50, Failures: 0, Errors: 50, Skipped: 0, Time elapsed: 0.961 sec <<< FAILURE!
[INFO] test_access_component_after_deletion_in_previous_system(com.artemis.WorldTest)  Time elapsed: 0.928 sec  <<< ERROR!
[INFO] com.google.gwt.core.ext.UnableToCompleteException: (see previous log entries)
[INFO] 	at com.google.gwt.junit.JUnitShell.compileForWebMode(JUnitShell.java:1095)
[INFO] 	at com.google.gwt.junit.JUnitShell.maybeCompileForWebMode(JUnitShell.java:1148)
[INFO] 	at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl2(CompileStrategy.java:183)
[INFO] 	at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl(CompileStrategy.java:113)
[INFO] 	at com.google.gwt.junit.SimpleCompileStrategy.maybeCompileModule(SimpleCompileStrategy.java:36)
[INFO] 	at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1313)
[INFO] 	at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1281)
[INFO] 	at com.google.gwt.junit.JUnitShell.runTest(JUnitShell.java:670)
[INFO] 	at com.google.gwt.junit.client.GWTTestCase.runTest(GWTTestCase.java:421)
[INFO] 	at junit.framework.TestCase.runBare(TestCase.java:127)
[INFO] 	at junit.framework.TestResult$1.protect(TestResult.java:106)
[INFO] 	at junit.framework.TestResult.runProtected(TestResult.java:124)
[INFO] 	at junit.framework.TestResult.run(TestResult.java:109)
[INFO] 	at junit.framework.TestCase.run(TestCase.java:118)
[INFO] 	at com.google.gwt.junit.client.GWTTestCase.run(GWTTestCase.java:247)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at junit.framework.TestSuite.runTest(TestSuite.java:208)
[INFO] 	at junit.framework.TestSuite.run(TestSuite.java:203)
[INFO] 	at org.codehaus.mojo.gwt.test.MavenTestRunner.doRun(MavenTestRunner.java:105)
[INFO] 	at junit.textui.TestRunner.start(TestRunner.java:172)
[INFO] 	at org.codehaus.mojo.gwt.test.MavenTestRunner.main(MavenTestRunner.java:63)
```

**Comment by DaanVanYperen** [comment]
see https://github.com/libgdx/libgdx/issues/5256

**Comment by DaanVanYperen** [comment]
Potentially can be solved with 'com.google.code.findbugs:jsr305:3.0.2:sources' package.

**Comment by DaanVanYperen** [comment]
Fixed by:
1. applying fix from https://stackoverflow.com/questions/27522728/unable-to-compile-the-gwt-module-while-trying-to-use-javax-annotation-nullable 
2. Depend test module on gwt 2.8.2.

Took the opportunity to test the build on AdoptOpenJDK 11, as well as Oracle JDK 8 and 10, seems to work fine. 

Got sidetracked a bit by the Jetty test launcher gwt-maven-plugin throwing a ClassNotFoundException sun.misc.GC, but that exception can be safely ignored. Should be fixed in 2.9 version of gwt-maven-plugin https://github.com/gwtproject/gwt/issues/9561

```
[INFO] [ERROR] jreLeakPrevention.gcDaemonFail
[INFO] java.lang.ClassNotFoundException: sun.misc.GC
[INFO] 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
[INFO] 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
[INFO] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
[INFO] 	at java.base/java.lang.Class.forName0(Native Method)
[INFO] 	at java.base/java.lang.Class.forName(Class.java:315)
[INFO] 	at com.google.gwt.dev.shell.jetty.JettyLauncher.jreLeakPrevention(JettyLauncher.java:899)
[INFO] 	at com.google.gwt.dev.shell.jetty.JettyLauncher.start(JettyLauncher.java:722)
[INFO] 	at com.google.gwt.dev.DevMode.doStartUpServer(DevMode.java:636)
[INFO] 	at com.google.gwt.dev.DevModeBase.startUp(DevModeBase.java:923)
[INFO] 	at com.google.gwt.junit.JUnitShell.getUnitTestShell(JUnitShell.java:708)
```

</details>
